### PR TITLE
Fix recursive reference semantics

### PR DESCRIPTION
**Pull request by marrony**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,7 @@
 - Rename `ValueOption` to `ToOption` in `IResult<T>` interface
 - Adds support for `@query` type
 - Adds support for user classes serialization
-- Adds support for new reference types in FaunaDB API 2.1
```
Comment (erickpintor) [inline CHANGELOG.md]: Did we ever release this structures?

Comment (marrony) [reply CHANGELOG.md]: No, I branched the version 1.3 and only release that version, 2.x.x was not released yet

Comment (marrony) [reply CHANGELOG.md]: I mean, version 1.x doesn't have the new recursive references

Comment (erickpintor) [reply CHANGELOG.md]: Ok then. I was just wondering if we need to do a separated release for this changes but, if they were not released yet we are good.

**Pull request merged**